### PR TITLE
Possibility to clean up a course

### LINKED ISSUE #595: Possibility to clean up a course
It would be good if there was a way to remove all other users from the course.
It would be practical if this could be done using roles:

- [ ] Delete all students
- [ ] Delete all students & tutors

### PR DESCRIPTION
**Pull request by Jonas960**
resolves #595

### CODE REVIEW

File: ws/src/main/scala/de/thm/ii/fbs/services/persistence/CourseRegistrationService.scala [hunk 1]
```diff
@@ -35,6 +35,24 @@ class CourseRegistrationService {
...
+      * Deregister all users with a specific role from a course.
+      * @param cid The course id
+      * @param role The role
+      * @return True if sucessfully deregistered
+      */
+    def deregisterRole(cid: Int, role: CourseRole.Value): Boolean =
+      1 == DB.update("DELETE FROM user_course WHERE course_id = ? AND course_role = ?", cid, role.id)
+
+  /**
+      * Deregister all users except the current user .
```
Comment (mxsph) [inline ws/src/main/scala/de/thm/ii/fbs/services/persistence/CourseRegistrationService.scala:48]:
```suggestion
      * Deregister all users except the current user.
```

File: web-gui/src/app/service/course-registration.service.ts [hunk 2]
```diff
@@ -45,4 +45,23 @@ export class CourseRegistrationService {
...
   }
+
+  /**
+   * De-register all user with a specific role from a course
+   * @param roleName Role name
+   * @param cid Course id
+   */
+
+  deregisterRole(cid: number, roleName: string): Observable<void> {
+    return this.http.put<void>(`/api/v1/courses/${cid}/deregisterrole`, {roleName: roleName});
```
Comment (mxsph) [inline web-gui/src/app/service/course-registration.service.ts:56]:
```suggestion
    return this.http.put<void>(`/api/v1/courses/${cid}/deregisterrole`, {roleName});
```

**Review by mxsph — CHANGES_REQUESTED** [review]
Looks good so far :+1: 

There are only a few small things that still should be adjusted:

- [x] You should add the two new api routes to the API specification ([swagger.yml](https://github.com/thm-mni-ii/feedbacksystem/blob/2770ef11a1b050cecd9289e9661112cf7c0c92b4/swagger.yml)).

**Comment by Jonas960** [comment]
@mxsph I have implemented your suggestions

**Review by mxsph — COMMENTED** [review]
- [x] The buttons should be displayed at the top (so that they do not disappear when there are too many users).
  grafik
- [x] I find the color of the buttons a bit too bright. I would change the `red` to the color of the header (`#405e9a`)
  grafik
- [x] The rights check in the backend has a part that is superfluous and there is a bug, which leads to the situation that everyone can remove all users. I have already fixed this (see 0b610009ca61b41dd6139944ef49a9a119ec15cc).

**Review by mxsph — APPROVED** [review]

**Review by Zitrone44 — APPROVED** [review]

**Pull request merged**